pkg/serviceprovider/scmprovider.go

```diff
@@ -0,0 +1,22 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package serviceprovider
+
+//ScmProvider is a marker interface for service providers, indicating that given SP is able to
+// perform all listed SCM-specific operations, such as file retrieving or others.
+type ScmProvider interface {
+	// GetDownloadFileCapability returns download file capability
+	GetDownloadFileCapability() DownloadFileCapability
```

Comment:
What if the `ServiceProvider` interface had this method and returned a pointer to `DownloadCapability`? Returning `nil` would mean the service provider doesn't support file download.

That way we would avoid the checks by type-casting and maybe also simplify the situation in the testsuite where you would be simply able to make the `TestServiceProvider` support or not support file download...

Reply:
To avoid unnecesary stubbing for the providers that didn't have SCM capabilities

Reply:
In that case, why the double indirection? What if ScmProvider directly had the `DownloadFile` method?

Reply:
Although to be fair, I'm still more a fan of just extending the `ServiceProvider` interface (and keeping the `DownloadCapability` so that we can easily distinguish the SPs that support file download from the ones that do not).